Relax Linux checks for the code works on any non-Windows platform.

--- cargo-crates/wgpu-hal-27.0.4/src/vulkan/adapter.rs.orig	2006-07-24 01:21:28 UTC
+++ cargo-crates/wgpu-hal-27.0.4/src/vulkan/adapter.rs
@@ -1207,7 +1207,7 @@ impl PhysicalDeviceProperties {
         // don't risk confusing JS by exceeding the range of a double.
         let is_nvidia = self.properties.vendor_id == crate::auxil::db::nvidia::VENDOR;
         let max_buffer_size =
-            if (cfg!(target_os = "linux") || cfg!(target_os = "android")) && !is_nvidia {
+            if (cfg!(all(unix, not(target_vendor = "apple")))) && !is_nvidia {
                 i32::MAX as u64
             } else {
                 1u64 << 52
--- cargo-crates/wgpu-hal-27.0.4/src/vulkan/instance.rs.orig	2006-07-24 01:21:28 UTC
+++ cargo-crates/wgpu-hal-27.0.4/src/vulkan/instance.rs
@@ -988,7 +988,7 @@ impl crate::Instance for super::Instance {
             exposed.info.device_type == wgt::DeviceType::DiscreteGpu
                 && exposed.info.vendor == db::nvidia::VENDOR
         });
-        if cfg!(target_os = "linux") && has_nvidia_dgpu && self.shared.has_nv_optimus {
+        if cfg!(unix) && has_nvidia_dgpu && self.shared.has_nv_optimus {
             for exposed in exposed_adapters.iter_mut() {
                 if exposed.info.device_type == wgt::DeviceType::IntegratedGpu
                     && exposed.info.vendor == db::intel::VENDOR
--- server-cli/src/main.rs.orig	2025-12-17 17:31:20 UTC
+++ server-cli/src/main.rs
@@ -74,14 +74,14 @@ fn main() -> io::Result<()> {
     // Load settings
     let settings = settings::Settings::load().ok_or(io::ErrorKind::Other)?;
 
-    #[cfg(any(target_os = "linux", target_os = "macos"))]
+    #[cfg(not(target_os = "windows"))]
     {
         for signal in &settings.shutdown_signals {
             let _ = signal_hook::flag::register(signal.to_signal(), Arc::clone(&shutdown_signal));
         }
     }
 
-    #[cfg(not(any(target_os = "linux", target_os = "macos")))]
+    #[cfg(target_os = "windows")]
     if !settings.shutdown_signals.is_empty() {
         tracing::warn!(
             "Server configuration contains shutdown signals, but your platform does not support \
--- server-cli/src/settings.rs.orig	2025-12-17 17:31:20 UTC
+++ server-cli/src/settings.rs
@@ -15,7 +15,7 @@ impl ShutdownSignal {
 }
 
 impl ShutdownSignal {
-    #[cfg(any(target_os = "linux", target_os = "macos"))]
+    #[cfg(not(target_os = "windows"))]
     pub fn to_signal(self) -> core::ffi::c_int {
         match self {
             Self::SIGUSR1 => signal_hook::consts::SIGUSR1,
@@ -48,7 +48,7 @@ impl Default for Settings {
             web_address: SocketAddr::from((Ipv4Addr::LOCALHOST, 14005)),
             web_chat_secret: None,
             ui_api_secret: None,
-            shutdown_signals: if cfg!(any(target_os = "linux", target_os = "macos")) {
+            shutdown_signals: if cfg!(not(target_os = "windows")) {
                 vec![ShutdownSignal::SIGUSR1]
             } else {
                 Vec::new()
--- voxygen/src/main.rs.orig	2025-12-17 17:31:20 UTC
+++ voxygen/src/main.rs
@@ -47,7 +47,7 @@ fn main() {
             cli::Commands::ListWgpuBackends => {
                 #[cfg(target_os = "windows")]
                 let backends = &["opengl", "dx12", "vulkan"];
-                #[cfg(target_os = "linux")]
+                #[cfg(not(any(target_os = "windows", target_os = "macos")))]
                 let backends = &["opengl", "vulkan"];
                 #[cfg(target_os = "macos")]
                 let backends = &["metal"];
